### ISSUE TITLE
[bug]: a height hint greater than 0 must be provided

### ISSUE THREAD

**Issue by robtex** [lightningnetwork/lnd#8383]
### Background

lnd refuses to start, complainding about hint must be greater than 0

### Your environment

* version of `lnd`
lnd version 0.17.3-beta commit=v0.17.3-beta

* which operating system (`uname -a` on *Nix)

Linux lndmain 4.4.0-206-generic #238-Ubuntu SMP Tue Mar 16 07:52:37 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

* version of `btcd`, `bitcoind`, or other backend

Bitcoin Core version v25.0.0

* any other relevant environment details

### Steps to reproduce

start lnd.
having the same issue on two lnd nodes (with different transactions)
also tried 
`lnd  --height-hint-cache-query-disable`

```
2024-01-13 09:54:44.129 [ERR] BTCN: Broadcast attempt failed: tx already confirmed: -27: Transaction already in block chain 
2024-01-13 09:54:44.130 [INF] LNWL: Inserting unconfirmed transaction 62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00 
2024-01-13 09:54:44.291 [INF] LNWL: 62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00: tx already confirmed 
2024-01-13 09:54:44.450 [ERR] UTXN: Failed to sweep first-stage HTLC (CLTV-delayed) output 62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00:0 
2024-01-13 09:54:44.451 [ERR] UTXN: Failed to sweep outputs at height=0: a height hint greater than 0 must be provided 
2024-01-13 09:54:44.451 [INF] SWPR: Sweeper shutting down 
2024-01-13 09:54:44.451 [INF] NTFN: Cancelling epoch notification, epoch_id=2 
2024-01-13 09:54:44.451 [INF] NTFN: New confirmation subscription: conf_id=1, txid=62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00, num_confs=6 height_hint=825608 
2024-01-13 09:54:44.451 [INF] HSWC: HtlcNotifier shutting down 
2024-01-13 09:54:44.451 [INF] PRNF: PeerNotifier shutting down 
2024-01-13 09:54:44.451 [INF] CHNF: ChannelNotifier shutting down 
2024-01-13 09:54:44.451 [INF] NTFN: bitcoind notifier shutting down 
2024-01-13 09:54:45.441 [INF] NTFN: Dispatching 6 confirmation notification for txid=62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00 
2024-01-13 09:54:45.442 [INF] NTFN: Stopping mempool notifier 
2024-01-13 09:54:45.442 [INF] HLCK: Health monitor shutting down 
2024-01-13 09:54:45.442 [ERR] LTND: Shutting down because error in main method: unable to start server: a height hint greater than 0 must be provided 
```

### Expected behaviour

lnd should start

### Actual behaviour

does not start

**Comment by Roasbeef** [comment]
So removing the height hint cache as is, means we won't have a height hint to pass, but other sub-systems want to verify that it's present (as they need to know the last point of something being unspent/unconfirmed). 

Are you able to start if you unset the arg? Or did you run into this issue then try to set the arg? Had you set the arg before? 

Might the node have had other patches applied? It's odd that the nursery is trying to sweep something at height zero. Is the chain backend syncing from scratch in the background? More `UTXN` logs a the `debug` log level can help here as well.

**Comment by robtex** [comment]
> So removing the height hint cache as is, means we won't have a height hint to pass, but other sub-systems want to verify that it's present (as they need to know the last point of something being unspent/unconfirmed).
> 
> Are you able to start if you unset the arg? Or did you run into this issue then try to set the arg? Had you set the arg before?

i got the problems on the nodes first, then got a tip to set the flag. never used nor seen that flag before.

> 
> Might the node have had other patches applied?

one of the nodes have had several patches before, but not for years. the other node i think never had any patches.

> It's odd that the nursery is trying to sweep something at height zero.

especially when they are already swept and confirmed

>  Is the chain backend syncing from scratch in the background? More `UTXN` logs a the `debug` log level can help here as well.

don't think it syncs from scratch, i think i would have noticed.

**Comment by robtex** [comment]
added 
```
debuglevel=UTXN=trace
```
but don't see much UTXN stuff yet.
```
2024-02-01 18:36:52.399 [INF] LTND: Waiting for chain backend to finish sync, start_height=828418
```
seems it does not start at 0 though

**Comment by robtex** [comment]
```
2024-02-01 18:33:31.899 [INF] LTND: Version: 0.17.3-beta commit=v0.17.3-beta, build=production, logging=default, debuglevel=UTXN=trace
2024-02-01 20:04:17.393 [INF] UTXN: UTXO nursery starting
2024-02-01 20:04:19.475 [INF] UTXN: (Re)-sweeping 1 heights below height=828430
2024-02-01 20:04:19.475 [DBG] UTXN: Attempting to sweep outputs at height=0
2024-02-01 20:04:19.476 [INF] UTXN: Attempting to graduate height=0: num_kids=0, num_babies=23
2024-02-01 20:04:19.476 [INF] UTXN: Publishing CLTV-delayed HTLC output using timeout tx (txid=62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00): (*wire.MsgTx)(0xc001469180)({
2024-02-01 20:04:19.952 [ERR] UTXN: Failed to sweep first-stage HTLC (CLTV-delayed) output 62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00:0
2024-02-01 20:04:19.952 [ERR] UTXN: Failed to sweep outputs at height=0: a height hint greater than 0 must be provided
```
there is all UTXN from one session with debuglevel=UTXN=trace , hope it helps. any more settings i can try?

**Comment by robtex** [comment]
tried with 0.17.4, basically no change
after two hours it tried again to reinsert an already confirmed transaction and stopped the startup process complaining about height hint not greater than 0
would be happy if i could either submit that hint manually, or if the code could decide it is better to continue than stop

still same issue with two nodes with 0.17.4

```
2024-02-28 18:03:07.221 [WRN] CRTR: Routing failure for local channel 667421150308007936 occurred 
2024-02-28 18:03:07.221 [WRN] CRTR: Routing failure for local channel 667421150308007936 occurred 
2024-02-28 18:03:07.221 [WRN] CRTR: Routing failure for local channel 667421150308007936 occurred 
2024-02-28 18:03:07.221 [WRN] CRTR: Routing failure for local channel 667421150308007936 occurred 
2024-02-28 18:03:07.221 [WRN] CRTR: Routing failure for local channel 667421150308007936 occurred 
2024-02-28 18:03:07.315 [INF] LTND: Channel backup proxy channel notifier starting 
2024-02-28 18:03:07.316 [INF] ATPL: Instantiating autopilot with active=false, max_channels=5, allocation=0.600000, min_chan_size=20000, max_chan_size=16777215, private=false, min_confs=1, conf_target=3 
2024-02-28 18:03:07.316 [INF] LTND: We're not running within systemd or the service type is not 'notify' 
2024-02-28 18:03:08.131 [INF] LTND: Waiting for chain backend to finish sync, start_height=832432 
2024-02-28 18:35:57.773 [INF] LNWL: Catching up block hashes to height 830000, this might take a while 
2024-02-28 18:35:57.774 [INF] LNWL: Done catching up block hashes 
2024-02-28 18:35:57.774 [INF] LNWL: Rescanned through block 000000000000000000011d55599ed27d7efca05f5849b755319c89eb2cffbc1f (height 830000) 
2024-02-28 19:29:31.767 [INF] LNWL: Catching up block hashes to height 832438, this might take a while 
2024-02-28 19:29:31.768 [INF] LNWL: Done catching up block hashes 
2024-02-28 19:29:31.768 [INF] LNWL: Finished rescan for 113822 addresses (synced to block 0000000000000000000321834d8bb9f62e1b74c971ad3621223799459330098d, height 832438) 
2024-02-28 19:29:33.807 [INF] LTND: Chain backend is fully synced (end_height=832438)! 
2024-02-28 19:29:33.808 [WRN] HLCK: check: disk space configured with 0 attempts, skipping it 
2024-02-28 19:29:33.809 [WRN] HLCK: check: tls configured with 0 attempts, skipping it 
2024-02-28 19:29:33.809 [INF] LNWL: SigPool starting 
2024-02-28 19:29:34.304 [INF] LNWL: 6e12001da5af8dae817eee0928c8c9d1be2a67a6a6be02769af9676c32c5b188: tx already confirmed 
2024-02-28 19:29:35.711 [INF] CHNF: ChannelNotifier starting 
2024-02-28 19:29:35.711 [INF] PRNF: PeerNotifier starting 
2024-02-28 19:29:35.711 [INF] NTFN: New block epoch subscription 
2024-02-28 19:29:35.711 [INF] BTCN: Broadcaster now active 
2024-02-28 19:29:35.711 [INF] HSWC: HtlcNotifier starting 
2024-02-28 19:29:35.711 [INF] SWPR: Sweeper starting 
2024-02-28 19:29:35.861 [INF] UTXN: UTXO nursery starting 
2024-02-28 19:29:35.861 [INF] NTFN: New block epoch subscription 
2024-02-28 19:29:37.623 [INF] UTXN: (Re)-sweeping 1 heights below height=832438 
2024-02-28 19:29:37.623 [DBG] UTXN: Attempting to sweep outputs at height=0 
2024-02-28 19:29:37.623 [INF] UTXN: Attempting to graduate height=0: num_kids=0, num_babies=23 
2024-02-28 19:29:37.623 [INF] UTXN: Publishing CLTV-delayed HTLC output using timeout tx (txid=62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00): (*wire.MsgTx)(0xc008e8fec0)({ 
 Version: (int32) 2, 
 TxIn: ([]*wire.TxIn) (len=1 cap=1) { 
  (*wire.TxIn)(0xc000a98cc0)({ 
   PreviousOutPoint: (wire.OutPoint) 81e84010f3eb56e10eb808b00711c46cf1a96368015f83a21223c979c7b846f9:1, 
   SignatureScript: ([]uint8) { 
   }, 
   Witness: (wire.TxWitness) (len=5 cap=5) { 
    ([]uint8) { 
    }, 
    ([]uint8) (len=72 cap=72) { 
     00000000  30 45 02 21 00 87 d8 f4  39 83 5a 5d 71 46 17 fb  |0E.!....9.Z]qF..| 
     00000010  b4 c1 5b aa fd 91 89 9d  5c 30 82 fe 58 61 94 6c  |..[.....\0..Xa.l| 
     00000020  dd 58 af b3 38 02 20 19  33 74 07 a1 b8 c8 6d 43  |.X..8. .3t....mC| 
     00000030  a1 49 24 e9 a0 08 8e 4e  d7 e0 94 de d3 99 e6 40  |.I$....N.......@| 
     00000040  67 98 f4 92 43 27 92 01                           |g...C'..| 
    }, 
    ([]uint8) (len=71 cap=71) { 
     00000000  30 44 02 20 0d b3 01 32  c5 c3 b9 97 a2 bd 66 a1  |0D. ...2......f.| 
     00000010  aa c5 2a 55 6a 42 1f 2d  30 4d 6f 51 0a 66 b6 d7  |..*UjB.-0MoQ.f..| 
     00000020  0d ae 4d 57 02 20 44 d0  f9 62 38 81 f2 8d be f4  |..MW. D..b8.....| 
     00000030  57 48 7e 0a 36 dd 7f 0c  30 fb c9 1c a4 9e ba 25  |WH~.6...0......%| 
     00000040  64 25 04 7b 4d 39 01                              |d%.{M9.| 
    }, 
    ([]uint8) { 
    }, 
    ([]uint8) (len=133 cap=133) { 
     00000000  76 a9 14 69 d9 29 69 40  47 52 24 bd 20 64 c3 52  |v..i.)i@GR$. d.R| 
     00000010  10 f7 44 08 b5 18 72 87  63 ac 67 21 03 41 0a f5  |..D...r.c.g!.A..| 
     00000020  bb e6 aa b1 37 6b 79 26  df 99 3d 7f 8d a7 6f 39  |....7ky&..=...o9| 
     00000030  90 4c 1c 21 02 59 fe 61  72 72 6c 66 4c 7c 82 01  |.L.!.Y.arrlfL|..| 
     00000040  20 87 64 75 52 7c 21 02  9c 84 44 ae a5 0f 39 df  | .duR|!...D...9.| 
     00000050  08 19 f6 99 d9 72 36 ee  b5 53 ab c6 e8 8d a6 32  |.....r6..S.....2| 
     00000060  5f 0e 2d 14 4a 1a 0f 3b  52 ae 67 a9 14 b9 33 f0  |_.-.J..;R.g...3.| 
     00000070  45 99 7c 8c 49 52 af 17  92 c8 a4 bd 01 5a 29 be  |E.|.IR.......Z).| 
     00000080  c4 88 ac 68 68                                    |...hh| 
    } 
   }, 
   Sequence: (uint32) 0 
  }) 
 }, 
 TxOut: ([]*wire.TxOut) (len=1 cap=1) { 
  (*wire.TxOut)(0xc008e4b640)({ 
   Value: (int64) 15072, 
   PkScript: ([]uint8) (len=34 cap=34) { 
    00000000  00 20 e2 f3 5d 12 f8 c8  d0 14 cd bb ab 65 65 c8  |. ..]........ee.| 
    00000010  12 05 67 e2 aa bd eb 89  ef 05 bb c7 9e 89 e9 d5  |..g.............| 
    00000020  1e 9d                                             |..| 
   } 
  }) 
 }, 
 LockTime: (uint32) 0 
}) 
 
2024-02-28 19:29:37.775 [ERR] BTCN: Broadcast attempt failed: tx already confirmed: -27: Transaction already in block chain 
2024-02-28 19:29:37.775 [INF] LNWL: Inserting unconfirmed transaction 62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00 
2024-02-28 19:29:37.930 [INF] LNWL: 62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00: tx already confirmed 
2024-02-28 19:29:38.080 [ERR] UTXN: Failed to sweep first-stage HTLC (CLTV-delayed) output 62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00:0 
2024-02-28 19:29:38.080 [ERR] UTXN: Failed to sweep outputs at height=0: a height hint greater than 0 must be provided 
2024-02-28 19:29:38.080 [INF] SWPR: Sweeper shutting down 
2024-02-28 19:29:38.081 [INF] NTFN: Cancelling epoch notification, epoch_id=2 
2024-02-28 19:29:38.081 [INF] NTFN: New confirmation subscription: conf_id=1, txid=62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00, num_confs=6 height_hint=832438 
2024-02-28 19:29:38.081 [INF] HSWC: HtlcNotifier shutting down 
2024-02-28 19:29:38.081 [INF] PRNF: PeerNotifier shutting down 
2024-02-28 19:29:38.081 [INF] CHNF: ChannelNotifier shutting down 
2024-02-28 19:29:38.081 [INF] NTFN: bitcoind notifier shutting down 
2024-02-28 19:29:39.096 [INF] NTFN: Dispatching 6 confirmation notification for txid=62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00 
2024-02-28 19:29:39.096 [INF] NTFN: Stopping mempool notifier 
2024-02-28 19:29:39.096 [INF] HLCK: Health monitor shutting down 
2024-02-28 19:29:39.096 [ERR] LTND: Shutting down because error in main method: unable to start server: a height hint greater than 0 must be provided 
2024-02-28 19:29:39.096 [INF] RPCS: Stopping RPC Server 
2024-02-28 19:29:39.096 [INF] RPCS: Stopping RouterRPC Sub-RPC Server 
2024-02-28 19:29:39.096 [INF] RPCS: Stopping VersionRPC Sub-RPC Server 
2024-02-28 19:29:39.096 [INF] RPCS: Stopping WatchtowerClientRPC Sub-RPC Server 
2024-02-28 19:29:39.112 [INF] LTND: Shutdown complete 

unable to start server: a height hint greater than 0 must be provided 
```

**Comment by ziggie1984** [comment]
Very strange behavior, was the channel a LND <=> LND channel ?

**Comment by ziggie1984** [comment]
Can you run the patch: https://github.com/lightningnetwork/lnd/pull/8524

This PR just prints out the HTLC information of all the pending HTLCs which need to be spent by the UTXO_Nursery. Analysing the transaction which causes the height hint to be 0 brought my attention to the fact that all the locktimes of the Second-Level Timeout transactions are 0, although they should only be 0 for the Success-Transaction:

https://mempool.space/tx/62ff2f1f063082090be8aecd42bba5ae29b4b1b1a00e1f13806b9d99f8087e00

This is a timeout transaction which has the expiry of 0, which normally should not happen. By running the patch we can get more information of the htlc which is written to your db.

**Comment by robtex** [comment]
> Very strange behavior, was the channel a LND <=> LND channel ?

the channels on both nodes seems to be to LND.

**Comment by robtex** [comment]
@ziggie1984 i tried your patch and guess you want this part:
```
2024-03-10 15:34:20.111 [INF] UTXN: UTXO nursery starting 
2024-03-10 15:34:20.111 [INF] NTFN: New block epoch subscription 
2024-03-10 15:34:20.442 [INF] UTXN: (Re)-sweeping 1 heights below height=834066 
2024-03-10 15:34:20.442 [DBG] UTXN: Attempting to sweep outputs at height=0 
2024-03-10 15:34:20.442 [INF] UTXN: Attempting to graduate height=0: num_kids=0, num_babies=4 
2024-03-10 15:34:20.442 [DBG] UTXN: Crib output: babyOutput: expiry=0, timeoutTx=020000000001019c3d3bb647290174b822ff5a92ab60624a0945d96f61a0ab8fd56dfcdffd8e7502000000000000000001351701000000000022002094fd5b3395c57a37104d49af026868c950bb9052211d803fa560d1af9c154eb10500473044022009a134c2d405b66761a2446b53b0c0757fd3266432795c04f41e2c731b2f587202201e8a1fa63ff1372ef62effd47617143a52d854f6fa576169e0cc3e9e7300f4ff0147304402201324ccc13fec7ab900dcc4855825ac38b51f0609d880355ad42b4f44fa4d34fd0220271666dcf8c1ca70e9e82a2de583d7ad6342fd26dd9367267c303ee0bd46044101008576a914151f84cfe9ae31e279b2e35de4d77fdf1aeda77a8763ac672102b950360c6260bde44391a4e01178d0060b6def2ed2c02e956d9d70ef133124d57c820120876475527c2102cc01696b3eb22d453a0aa0788b2cff738996ea5c554287136cfc321290c1fac152ae67a9149934b68fc95bea23d23fe0305b522810e568e63d88ac686800000000, kidOutput=kidOutput: blocksToMaturity=144, absoluteMaturity=0, breachedOutput=breachedOutput: witnessType=HtlcOfferedTimeoutSecondLevel, amt=71477, outpoint=da2bffef57fd0ea5520d6cc2b53905cdf56735d702b65dd462f025c58f92490f:0, confHeight=0 
2024-03-10 15:34:20.442 [DBG] UTXN: Crib output: babyOutput: expiry=0, timeoutTx=020000000001019c3d3bb647290174b822ff5a92ab60624a0945d96f61a0ab8fd56dfcdffd8e7501000000000000000001eeb200000000000022002094fd5b3395c57a37104d49af026868c950bb9052211d803fa560d1af9c154eb105004830450221008acfebdafe1d64a7eb4789750e6306839a79f2f889517e1beead4d81337371620220758af7d20b9a2094a96a500d36b71b8a4c0dbeb9acf3815c7a947084611096760147304402203fac722814e9558329da224a5c6b10b8ec44ab9a427b540e96e31c11aa2831020220739d8dd1a0add2be831e38f750b15773ff23b887dba7c2a3b22b5058b309175601008576a914151f84cfe9ae31e279b2e35de4d77fdf1aeda77a8763ac672102b950360c6260bde44391a4e01178d0060b6def2ed2c02e956d9d70ef133124d57c820120876475527c2102cc01696b3eb22d453a0aa0788b2cff738996ea5c554287136cfc321290c1fac152ae67a914cb40a044188223e80851165ba62800e341d9cd2588ac686800000000, kidOutput=kidOutput: blocksToMaturity=144, absoluteMaturity=0, breachedOutput=breachedOutput: witnessType=HtlcOfferedTimeoutSecondLevel, amt=45806, outpoint=92087dff6457c5349cd77c39a1f5be4fcd32f8c1b514f3411920ab5339430e29:0, confHeight=0 
2024-03-10 15:34:20.443 [DBG] UTXN: Crib output: babyOutput: expiry=0, timeoutTx=020000000001019c3d3bb647290174b822ff5a92ab60624a0945d96f61a0ab8fd56dfcdffd8e7500000000000000000001cb7f00000000000022002094fd5b3395c57a37104d49af026868c950bb9052211d803fa560d1af9c154eb1050047304402203f64dd6000402eaba18ddf2472aa781cd5ce73ea30162417272ba1f9ee3e361b022053b624d75e48c3cc20232a74401b8728ea036f226e0526ed4a0147ecfaaf710601483045022100ff25b4622dd52d1e06c4f3849db1ee57be491456a4d682dede78671562b70d2d022050b6aabd78fa941568214f68ca9266e10f55c8382099a74212d97debc978db5001008576a914151f84cfe9ae31e279b2e35de4d77fdf1aeda77a8763ac672102b950360c6260bde44391a4e01178d0060b6def2ed2c02e956d9d70ef133124d57c820120876475527c2102cc01696b3eb22d453a0aa0788b2cff738996ea5c554287136cfc321290c1fac152ae67a91432d82dbe64484894960f83a16c149c239f26a01588ac686800000000, kidOutput=kidOutput: blocksToMaturity=144, absoluteMaturity=0, breachedOutput=breachedOutput: witnessType=HtlcOfferedTimeoutSecondLevel, amt=32715, outpoint=7970f5660fa5af13d1678976f66efb5958194102978842206d6b4dbd2ac2b7bd:0, confHeight=0 
2024-03-10 15:34:20.443 [DBG] UTXN: Crib output: babyOutput: expiry=0, timeoutTx=020000000001019c3d3bb647290174b822ff5a92ab60624a0945d96f61a0ab8fd56dfcdffd8e7503000000000000000001361701000000000022002094fd5b3395c57a37104d49af026868c950bb9052211d803fa560d1af9c154eb105004830450221008b96bb27827efec62dd3496f51fa60fa6242bc68ab1a944f189b28cae9cf9a6d02206e25e3026e31cb0f1fef75572a3bd186128c1390786725b2c7a4bc6f6d3b840b0148304502210090247d7e91179f487c74176cf33c9f42efd38c0f68689e5ca5685e78b9eae78602207a3ae49d160fe3d08dbf8bb9658c12e568ceda20583f93ef69c6a9cfe9ea0e1a01008576a914151f84cfe9ae31e279b2e35de4d77fdf1aeda77a8763ac672102b950360c6260bde44391a4e01178d0060b6def2ed2c02e956d9d70ef133124d57c820120876475527c2102cc01696b3eb22d453a0aa0788b2cff738996ea5c554287136cfc321290c1fac152ae67a9143b788e941926604a7221e437124a584280be12a188ac686800000000, kidOutput=kidOutput: blocksToMaturity=144, absoluteMaturity=0, breachedOutput=breachedOutput: witnessType=HtlcOfferedTimeoutSecondLevel, amt=71478, outpoint=1e2c311bee786a57fde8594ffe151e4134c2dd2a6175954105f522d96032b8f7:0, confHeight=0 
2024-03-10 15:34:20.443 [INF] UTXN: Publishing CLTV-delayed HTLC output using timeout tx (txid=da2bffef57fd0ea5520d6cc2b53905cdf56735d702b65dd462f025c58f92490f): (*wire.MsgTx)(0xc001f1fd40)({ 
 Version: (int32) 2, 
 TxIn: ([]*wire.TxIn) (len=1 cap=1) { 
  (*wire.TxIn)(0xc000cc87e0)({ 
   PreviousOutPoint: (wire.OutPoint) 758efddffc6dd58faba0616fd945094a6260ab925aff22b874012947b63b3d9c:2, 
   SignatureScript: ([]uint8) { 
   }, 
   Witness: (wire.TxWitness) (len=5 cap=5) { 
    ([]uint8) { 
    }, 
    ([]uint8) (len=71 cap=71) { 
     00000000  30 44 02 20 09 a1 34 c2  d4 05 b6 67 61 a2 44 6b  |0D. ..4....ga.Dk| 
     00000010  53 b0 c0 75 7f d3 26 64  32 79 5c 04 f4 1e 2c 73  |S..u..&d2y\...,s| 
     00000020  1b 2f 58 72 02 20 1e 8a  1f a6 3f f1 37 2e f6 2e  |./Xr. ....?.7...| 
     00000030  ff d4 76 17 14 3a 52 d8  54 f6 fa 57 61 69 e0 cc  |..v..:R.T..Wai..| 
     00000040  3e 9e 73 00 f4 ff 01                              |>.s....| 
    }, 
    ([]uint8) (len=71 cap=71) { 
     00000000  30 44 02 20 13 24 cc c1  3f ec 7a b9 00 dc c4 85  |0D. .$..?.z.....| 
     00000010  58 25 ac 38 b5 1f 06 09  d8 80 35 5a d4 2b 4f 44  |X%.8......5Z.+OD| 
     00000020  fa 4d 34 fd 02 20 27 16  66 dc f8 c1 ca 70 e9 e8  |.M4.. '.f....p..| 
     00000030  2a 2d e5 83 d7 ad 63 42  fd 26 dd 93 67 26 7c 30  |*-....cB.&..g&|0| 
     00000040  3e e0 bd 46 04 41 01                              |>..F.A.| 
    }, 
    ([]uint8) { 
    }, 
    ([]uint8) (len=133 cap=133) { 
     00000000  76 a9 14 15 1f 84 cf e9  ae 31 e2 79 b2 e3 5d e4  |v........1.y..].| 
     00000010  d7 7f df 1a ed a7 7a 87  63 ac 67 21 02 b9 50 36  |......z.c.g!..P6| 
     00000020  0c 62 60 bd e4 43 91 a4  e0 11 78 d0 06 0b 6d ef  |.b`..C....x...m.| 
     00000030  2e d2 c0 2e 95 6d 9d 70  ef 13 31 24 d5 7c 82 01  |.....m.p..1$.|..| 
     00000040  20 87 64 75 52 7c 21 02  cc 01 69 6b 3e b2 2d 45  | .duR|!...ik>.-E| 
     00000050  3a 0a a0 78 8b 2c ff 73  89 96 ea 5c 55 42 87 13  |:..x.,.s...\UB..| 
     00000060  6c fc 32 12 90 c1 fa c1  52 ae 67 a9 14 99 34 b6  |l.2.....R.g...4.| 
     00000070  8f c9 5b ea 23 d2 3f e0  30 5b 52 28 10 e5 68 e6  |..[.#.?.0[R(..h.| 
     00000080  3d 88 ac 68 68                                    |=..hh| 
    } 
   }, 
   Sequence: (uint32) 0 
```

**Comment by ziggie1984** [comment]
So the problem in your case is, that HTLCs where created with a LockTime (expiry) of 0. How that happened I cannot explain so far normally it is guarded by this function:

https://github.com/lightningnetwork/lnd/blob/5cb4811e86ed1607c1c9671756d7276256e2ec44/htlcswitch/link.go#L2816-L2818

Specifically: https://github.com/lightningnetwork/lnd/blob/5cb4811e86ed1607c1c9671756d7276256e2ec44/htlcswitch/link.go#L2910-L2919 

where we make sure the htlc timeout is greater than the current height + a delta. So prob. the LND you run when this happened was somewhere in the `v0.6-beta` to `v0.5-beta` range. I wonder whether this node ever run successfully after 2019 where the FCs happened?

To sum it up, your case should never happen with LND. I think the best solution for you is to run a patched version to recover your funds and then close the node down. I let you know which patch to run to recover your funds.

**Comment by ziggie1984** [comment]
So I added a code part into #8524 which now tries to fill in the missing information when the height_hint is zero by querying the bitcoind backend. This will only work with a full node (not pruned bitcoind node). With this change your lnd node should start up fine. make sure that all the relevant channels are fully resolved before switching versions again.

I am not sure about adding this kind of exception into the main code, because from my understanding a height_hint of 0 cannot happen currently because we guard against it as shown above.

Feel free to reach out if you have any questions @robtex

**Comment by ziggie1984** [comment]
So I think you run lnd-0.5-beta when this problem occured, because there a Outgoing HTLC of 0, would have been allowed:

https://github.com/lightningnetwork/lnd/blob/3b2c807288b1b7f40d609533c1e96a510ac5fa6d/htlcswitch/link.go#L1953-L1965

So maybe a bad implementation of pathfinding caused this, but in the newest versions of LND starting from lnd-0.5.1-beta this should not be a problem anymore.

**Comment by saubyk** [comment]
Closing the issue based on the last comment. Feel free to reopen if the issue persists.
